elastic/kibana
# [Response Ops] [Alerting] Refactor alerting task runner - single function for scheduling actions

Resolves https://github.com/elastic/kibana/issues/131548

## Summary
Combines the logic inside
```
await Promise.all(
        alertsWithExecutableActions.map(
          ([alertId, alert]: [string, Alert<InstanceState, InstanceContext>]) =>
            this.executeAlert(alertId, alert, executionHandler, ruleRunMetricsStore)
        )
      );

await scheduleActionsForRecoveredAlerts<
        InstanceState,
        InstanceContext,
        RecoveryActionGroupId
      >({
```
into a single library function.


### Checklist



### To Verify
- Verify that your actions still run correctly on your rules
- Verify that the tests pass
